### PR TITLE
Add support for getting logs from a Kubernetes Pod

### PR DESCRIPTION
**Pull request by fabianvf**
##### SUMMARY

Adds the `k8s_log` module, which allows you to pull logs from any resource that supports the `/log` endpoint (mostly just Pods I think).

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

k8s
k8s_log

### CODE REVIEW

File: lib/ansible/modules/clustering/k8s/k8s_log.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  - The text log of the object
+  returned: success
+'''
+
+
+from ansible.module_utils.k8s.common import KubernetesAnsibleModule, AUTH_ARG_SPEC
+import copy
+
+
+class KubernetesFactsModule(KubernetesAnsibleModule):
```
Comment (djzager) [inline lib/ansible/modules/clustering/k8s/k8s_log.py]: This class name doesn't look write. Copypasta?

File: lib/ansible/modules/clustering/k8s/k8s_log.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    description:
+    - Use to specify an object name.  Use in conjunction with I(api_version), I(kind) and I(namespace) to identify a
+      specific object.
+  namespace:
+    description:
+    - Use to specify an object namespace. Use in conjunction with I(api_version), I(kind), and I(name)
+      to identify a specfic object.
+  container:
+    description:
+    - Use to specify the container within a pod to grab the log from
```
Comment (djzager) [inline lib/ansible/modules/clustering/k8s/k8s_log.py]: Does this not being required mean that you will grab the log from the first container? Maybe you could specify the behavior when it's not included.

**Review by djzager — APPROVED** [review]
Not that my vote means much. But this LGTM.

**Comment by maxamillion** [comment]
I'm not entirely sure this adds any "value" over calling the REST API directly, which goes against documented Ansible development guidelines. https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_best_practices.html

**Comment by fabianvf** [comment]
There's significant complexity to authenticating to the API, doing it generically in pure Ansible would be really dirty. this module uses the same authentication methods as the other k8s modules, which can load certs, incluster configuration, kubeconfigs (which can include executing binaries for aws/gke etc). I think removing this complexity constitutes value over calling the API directly.

File: lib/ansible/modules/clustering/k8s/k8s_log.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        ))
+
+    @property
+    def argspec(self):
+        args = copy.deepcopy(AUTH_ARG_SPEC)
+        args.update(
+            dict(
+                kind=dict(default='Pod'),
+                api_version=dict(default='v1', aliases=['api', 'version']),
+                name=dict(),
```
Comment (willthames) [inline lib/ansible/modules/clustering/k8s/k8s_log.py:139]: I think either name should be mandatory or this should take a label selector.

**Comment by fabianvf** [comment]
ping @willthames I added the ability to use a label_selector, as well as grab label_selectors from the spec of an object that define selectors (like `Deployments`, `StatefulSets` etc).

**Comment by jtaleric** [comment]
Curious the state of this, what is holding things up?

**Comment by aakarshg** [comment]
@willthames Is this good to go, or any other changes required? Don't mean to rush you, but this module looks like a valuable addition and more of a necessity than a good to have.

**Comment by geerlingguy** [comment]
@fabianvf - The Kubernetes-related modules and plugins have been moved to a new `community.kubernetes` collection, where further development and module additions (like this one) will go. Would you be willing/able to create a new PR adding this module to that project?

Collection repo: https://github.com/ansible-collections/kubernetes

**Pull request closed without merging**